misc/test-ca/demoCA/test_client.crt

```diff
@@ -0,0 +1,17 @@
+-----BEGIN CERTIFICATE-----
```

Comment:
I might have been unclear in [my previous comment](https://github.com/h2o/h2o/pull/2621#discussion_r599091940).

misc/test-ca/demoCA contains the state of the test CA. It is governed by the `openssl ca` command, and is updated as the test CA issues new certificates. I believe you used the command when generating `test_client.crt`.

The problem here are:
* We cannot add arbitrary files to misc/test-ca/demoCA, because the directory is governed by `openssl ca` command.
* When issuing new certificates, the changes made by `openssl ca` command have to be committed, otherwise we'd have inconsistencies between the certificates being issued.

I think it would be a good idea to:
* move test_client.key and test_client.crt to t/assets (as they are used only by tests), and 
* commit the changes made to misc/test-ca/demoCA when test_client.crt was issued, so that the state of the CA reflects the fact that test_client.crt has been issued. If you have lost those states, I think you can regenerate test-client.crt.

Reply:
I will move test_client.key and .cert to t/assets. I did not use "openssl ca" command to generate those certs, instead I used "openssl req" and "openssl x509" to generate these.